api/tests/unit/infrastructure/repositories/correction-repository_test.js

```diff
@@ -81,25 +72,71 @@ describe('Unit | Repository | correction-repository', function() {
+      });
+
+      it('should return a correction with the solution to display', async function() {
+        // given
+        const expectedCorrection = new Correction({
+          id: 'recwWzTquPlvIl4So',
+          solution: '1, 5',
+          hints: expectedHints,
+          tutorials: expectedTutorials,
+          learningMoreTutorials: expectedLearningMoreTutorials,
+        });
+        challengeDataObject = ChallengeLearningContentDataObjectFixture({
+          skillIds: ['recIdSkill001', 'recIdSkill002', 'recIdSkill003'],
+          solution: '',
+          solutionToDisplay: '1, 5',
+        });
+        challengeDatasource.get.resolves(challengeDataObject);
 
         // when
         promise = correctionRepository.getByChallengeId({ challengeId: recordId, userId, locale });
-      });
 
-      it('should return a correction with the solution', function() {
         // then
-        return promise.then((result) => {
-          expect(result).to.be.an.instanceof(Correction);
-          expect(result).to.deep.equal(expectedCorrection);
-          expect(challengeDatasource.get).to.have.been.calledWith(recordId);
+        const result = await promise;
+        expect(result).to.be.an.instanceof(Correction);
+        expect(result).to.deep.equal(expectedCorrection);
+        expect(challengeDatasource.get).to.have.been.calledWith(recordId);
+      });
+
+      it('should return a correction with the solution', async function() {
+        // given
+        const expectedCorrection = new Correction({
+          id: 'recwWzTquPlvIl4So',
+          solution: '2',
+          hints: expectedHints,
+          tutorials: expectedTutorials,
+          learningMoreTutorials: expectedLearningMoreTutorials,
         });
+        challengeDataObject = ChallengeLearningContentDataObjectFixture({
+          skillIds: ['recIdSkill001', 'recIdSkill002', 'recIdSkill003'],
+          solutionToDisplay: '',
+          solution: '2',
+        });
+        challengeDatasource.get.resolves(challengeDataObject);
+
+        // when
+        promise = correctionRepository.getByChallengeId({ challengeId: recordId, userId, locale });
+
+        // then
+        const result = await promise;
+        expect(result).to.be.an.instanceof(Correction);
+        expect(result).to.deep.equal(expectedCorrection);
+        expect(challengeDatasource.get).to.have.been.calledWith(recordId);
       });
 
       it('should return the correction with hints that are validated', function() {
+        // given
+        challengeDataObject = ChallengeLearningContentDataObjectFixture({ skillIds: ['recIdSkill001', 'recIdSkill002', 'recIdSkill003'] });
+        challengeDatasource.get.resolves(challengeDataObject);
+
+        // when
+        promise = correctionRepository.getByChallengeId({ challengeId: recordId, userId, locale });
+
```

Comment:
```suggestion
        const result = await correctionRepository.getByChallengeId({ challengeId: recordId, userId, locale });
        
        // then
        return result.hints.forEach((hint) => expect(hint).to.be.an.instanceof(Hint));
```
**Suggestion** : Si pas de raison particulière d'attendre avant de resolve la `promise`